src/Compilers/CSharp/Portable/Compiler/MethodCompiler.cs

```diff
@@ -1368,6 +1368,33 @@ private BoundStatement ChainImplicitStructConstructor(MethodSymbol methodSymbol,
                     Debug.Assert(!diagnostics.HasAnyErrors());
                 }
 
+                DynamicAnalysisMethodBodyData dynamicAnalysisDataOpt = null;
+                if (moduleBuilder.EmitOptions.EmitDynamicAnalysisData)
+                {
+                    Debug.Assert(debugDocumentProvider != null);
+
+                    // TODO: collect information from bound trees
+                    if (method.Name == "Main")
```

Comment:
Nope, this is valid product code! :) 
I needed to test some data. There is no easy way to add that data until John implements the part that builds the spans from bound trees.